File: parciales/ED1-Parcial1-2021-1--001/codigo/Subcadenas.java
class Subcadenas{
public static int solve(String s, int k){
  if(s.length() < k) return 0;
  int[] count = new int[26];
  for(int i = 0; i < k; ++i){
    assert s.charAt(i) - 97 >= 0;
    count[s.charAt(i) - 97]++;
  }
  int ans = check(count) ? 1: 0;
  for(int i = k; i < s.length(); ++i){
    count[s.charAt(i - k) - 97] --;
    count[s.charAt(i) - 97] = count[s.charAt(i) - 97] + 1;
    if(check(count)){
      ans++;
    }
  }
  return ans;
}
public static boolean check(int[] count){
  for(int i = 0; i < 26; ++i){
    if(count[i] > 1) return false;
  }
  return true;
}
public static void main(String[] args){
  System.out.println(solve("andamasqueyo",5));
  System.out.println(solve("casa",5));
}
}

File: parciales/ED1-Parcial1-2021-1--001/codigo/Subcadenas.py
import numpy as np
def solve(s, k):
  if(len(s) < k): 
    return 0
  count = np.zeros(26)
  for i in range (0,k):
    count[ord(s[i]) - 97] += 1
  ans = 1 if check(count) else 0
  for i in range (k, len(s)): 
    count[ord(s[i - k]) - 97] -= 1
    count[ord(s[i]) - 97] = count[ord(s[i]) - 97] + 1
    if check(count):
      ans += 1
  return ans

def check(count):
  for i in range(0,26):
    if count[i] > 1:
     return False
  return True

def __main__():
  print(solve("andamasqueyo",5))
  print(solve("casa",5))

__main__()
